src/app/products/manage-tax-groups/create-tax-group/create-tax-group.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, FormArray } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ProductsService } from 'app/products/products.service';
+import { MatDialog } from '@angular/material';
+import { CancelDialogComponent } from 'app/shared/cancel-dialog/cancel-dialog.component';
+import { DatePipe } from '@angular/common';
+
+@Component({
+  selector: 'mifosx-create-tax-group',
+  templateUrl: './create-tax-group.component.html',
+  styleUrls: ['./create-tax-group.component.scss']
+})
+export class CreateTaxGroupComponent implements OnInit {
+  TaxGroupForm: FormGroup;
+    /** tax Component Data. */
+    taxComponentData: any;
+
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe,
+              public dialog: MatDialog,
+              private productService: ProductsService) {}
+
+  ngOnInit() {
+    // Create initial form
+    this.createTaxGroupForm();
+    this.productService.getTaxComponents().subscribe((data: any) => {
+      this.taxComponentData = data;
+    });
+  }
+
+  // Creates tax group form
+  createTaxGroupForm() {
+    this.TaxGroupForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'taxComponents': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Gets the taxcomponent form array.
+   * @returns {FormArray} taxcomponent form array.
+   */
+  get taxComponents(): FormArray {
+    return this.TaxGroupForm.get('taxComponents') as FormArray;
+  }
+
+
+  createTaxcomponentForm(): FormGroup {
```

Comment:
Missing comment
```suggestion
  createTaxComponentForm(): FormGroup {
```
Change other instances accordingly.